### PR TITLE
Switch to Gradle & OpenRewrite 8.0

### PR DESCRIPTION
**Pull request by timtebeek**
## What's changed?
Switch to Gradle & OpenRewrite 8.0

## What's your motivation?
Re-adopt cucumber-jvm-migration to make it fit in with OpenRewrite release schedule.
As a separate module to reduce the weight of rewrite-testing-frameworks & simplify access controls.

## Anything in particular you'd like reviewers to focus on?
JavaTemplate use of contextSensitive & Cursor.

## Have you considered any alternatives or workarounds?
Including these in rewrite-testing-frameworks would inflate the dependencies there again, and set precedent of owner-less multi library modules.

## Any additional context
These recipes were previously in rewrite-testing-frameworks, where I contributed them before we briefly considered publishing these from the cucumber orginisation, where I'm also a contributor.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ auto-formatter on affected files
- [x] I've updated the documentation (if applicable)

### CODE REVIEW

File: src/main/java/org/openrewrite/cucumber/jvm/CucumberAnnotationToSuite.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                classDecl = classDecl
+                        .withLeadingAnnotations(ListUtils.map(classDecl.getLeadingAnnotations(), ann -> {
+                            if (cucumberAnnoMatcher.matches(ann)) {
+                                String code = "@SelectClasspathResource(\"#{}\")";
+                                return JavaTemplate.builder(code)
+                                        .contextSensitive()
+                                        .javaParser(javaParserSupplier)
+                                        .imports(SELECT_CLASSPATH_RESOURCE)
+                                        .build()
+                                        .apply(getCursor(), ann.getCoordinates().replace(), classDeclPath);
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/cucumber/jvm/CucumberAnnotationToSuite.java:98]: This currently fails with
```
java.lang.AssertionError: Failed to parse sources or run recipe

	at org.openrewrite.test.RewriteTest.lambda$defaultExecutionContext$11(RewriteTest.java:544)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.handleError(RecipeScheduler.java:266)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.lambda$editSources$4(RecipeScheduler.java:210)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.lambda$mapForRecipeRecursively$5(RecipeScheduler.java:292)
	at org.openrewrite.internal.InMemoryLargeSourceSet.lambda$edit$0(InMemoryLargeSourceSet.java:61)
	at org.openrewrite.internal.ListUtils.lambda$map$0(ListUtils.java:166)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:147)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:166)
	at org.openrewrite.internal.InMemoryLargeSourceSet.edit(InMemoryLargeSourceSet.java:60)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.mapForRecipeRecursively(RecipeScheduler.java:285)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.editSources(RecipeScheduler.java:166)
	at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:75)
	at org.openrewrite.Recipe.run(Recipe.java:300)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:328)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:130)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:125)
	at org.openrewrite.cucumber.jvm.CucumberAnnotationToSuiteTest.shouldReplaceCucumberAnnotationWithSuiteWithSelectedClasspathResource(CucumberAnnotationToSuiteTest.java:39)
Caused by: org.openrewrite.internal.RecipeRunException: Exception while visiting project file 'com/example/app/CucumberJava8Definitions.java (in CucumberJava8Definitions)', caused by: java.lang.ClassCastException: class org.openrewrite.java.tree.J$ClassDeclaration cannot be cast to class org.openrewrite.java.tree.J$Annotation (org.openrewrite.java.tree.J$ClassDeclaration and org.openrewrite.java.tree.J$Annotation are in unnamed module of loader 'app'), at org.openrewrite.cucumber.jvm.CucumberAnnotationToSuite$ExecutionContextJavaIsoVisitor.lambda$visitClassDeclaration$0(CucumberAnnotationToSuite.java:98)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:330)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:362)
	at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10(JavaVisitor.java:468)
	at org.openrewrite.internal.ListUtils.lambda$map$0(ListUtils.java:166)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:147)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:166)
	at org.openrewrite.java.JavaVisitor.visitCompilationUnit(JavaVisitor.java:468)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:113)
	at org.openrewrite.java.JavaIsoVisitor.visitCompilationUnit(JavaIsoVisitor.java:30)
	at org.openrewrite.java.tree.J$CompilationUnit.acceptJava(J.java:1520)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:279)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:185)
	at org.openrewrite.Preconditions$1.visit(Preconditions.java:51)
	at org.openrewrite.Preconditions$1.visit(Preconditions.java:31)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.lambda$editSources$3(RecipeScheduler.java:196)
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:147)
	at org.openrewrite.table.RecipeRunStats.recordEdit(RecipeRunStats.java:54)
	at org.openrewrite.RecipeScheduler$RecipeRunCycle.lambda$editSources$4(RecipeScheduler.java:193)
	... 14 more
Caused by: java.lang.ClassCastException: class org.openrewrite.java.tree.J$ClassDeclaration cannot be cast to class org.openrewrite.java.tree.J$Annotation (org.openrewrite.java.tree.J$ClassDeclaration and org.openrewrite.java.tree.J$Annotation are in unnamed module of loader 'app')
	at org.openrewrite.cucumber.jvm.CucumberAnnotationToSuite$ExecutionContextJavaIsoVisitor.lambda$visitClassDeclaration$0(CucumberAnnotationToSuite.java:98)
	at org.openrewrite.internal.ListUtils.lambda$map$0(ListUtils.java:166)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:147)
	at org.openrewrite.internal.ListUtils.map(ListUtils.java:166)
	at org.openrewrite.cucumber.jvm.CucumberAnnotationToSuite$ExecutionContextJavaIsoVisitor.visitClassDeclaration(CucumberAnnotationToSuite.java:90)
	at org.openrewrite.cucumber.jvm.CucumberAnnotationToSuite$ExecutionContextJavaIsoVisitor.visitClassDeclaration(CucumberAnnotationToSuite.java:68)
	at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava(J.java:1217)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:279)
	... 32 more
```

Comment (knutwannheden) [reply src/main/java/org/openrewrite/cucumber/jvm/CucumberAnnotationToSuite.java:98]: Try replacing `getCursor()` with `new Cursor(new Cursor(getCursor().getParent(), classDecl), ann)`.

Comment (timtebeek) [reply src/main/java/org/openrewrite/cucumber/jvm/CucumberAnnotationToSuite.java:98]: Tried this for a bit, but did not get it to work yet. Not high priority for now, so will look into some other issues first.

Comment (timtebeek) [reply src/main/java/org/openrewrite/cucumber/jvm/CucumberAnnotationToSuite.java]: Took a step back and fixed it in https://github.com/openrewrite/rewrite-cucumber-jvm/pull/4/commits/4fd6f5696679dc9e2e03969eba3450c26e21492d

**Pull request merged**